src/api/firebase.js

```diff
@@ -29,6 +29,50 @@ export function streamListItems(listId, handleSuccess) {
 	return onSnapshot(listCollectionRef, handleSuccess);
 }
 
+/**
+ * comparePurchaseUrgency CURRENT FUNCTIONS:
+ * - sort for inactive items last, then
+ * - sort for overdue items to be listed at the top that aren't inactive, then
+ * - sort to most urgent date next purchased, then
+ * - sort to earliest alphabet name if date next purchased is the same
+ */
+export function comparePurchaseUrgency(item1, item2) {
+	//Declare current date
+	const currentDate = new Date();
+
+	//Returns the days until next purchase between currentDate and dateNextPurchased for item argument
+	function getDaysUntilNextPurchase(item) {
+		return getDaysBetweenDates(currentDate, item.dateNextPurchased.toDate());
+	}
+
+	//Inactive item conditional
+	if (getDaysUntilNextPurchase(item1) >= 60) {
+		return 1;
+	}
+	//Overdue item conditional
+	if (currentDate > item1.dateNextPurchased.toDate()) {
+		return -1;
+	} else if (currentDate > item2.dateNextPurchased.toDate()) {
+		return 1;
+	}
+	//Conditionals for days until next purchase and alphabetized if same number of days
+	if (getDaysUntilNextPurchase(item1) < getDaysUntilNextPurchase(item2)) {
+		return -1;
+	} else if (
+		getDaysUntilNextPurchase(item1) > getDaysUntilNextPurchase(item2)
+	) {
+		return 1;
+	} else {
+		// if days until next purchase between item1 and item2 are the same,
+		// we return the item earlier in the alphabet
+		if (item1.name < item2.name) {
```

Comment:
before reading the comments, i would think the list was not sorted at all:

<img width="300" alt="Screen Shot 2023-02-16 at 2 59 31 PM" src="https://user-images.githubusercontent.com/44907238/219507163-0ef22417-98a7-4bfa-b123-a936f26aee97.png">

Reply:
Good catch!!! We just fixed this sort - it was an overlap with inactive and overdue sorting